File: statSim.py
from math import sqrt
import numpy as np
from matplotlib import pyplot as plt
N = 1000
displacement=[]
for j in range(0,N):
    thita = np.random.randint(361, size=N)
    x = np.cos(thita)
    y = np.sin(thita)
    for i in range(0, N):
        x[i] = x[i-1] + x[i]
        y[i] = y[i-1] + y[i]
    displacement.append(sqrt(x[N-1]*x[N-1]+y[N-1]*y[N-1]))
print(f"Max distance is {max(displacement)} and average distance is {np.average(displacement)}")
plt.plot(x,y)
plt.show()

File: statsim.c
#include<stdio.h>
#include<math.h>
#include<stdlib.h>
#include<time.h>
#define N 1000
int main(){
	float x,y,displacement[N];
	int thita;
	float max=0, average=0;
	srand(time(0));
	for(int j=0; j<N; j++){
		x=y=0;
		for(int i=0; i<N;i++){
			thita = rand()%361;
			x+=cos(thita);
			y+=sin(thita);
		}
		displacement[j] = sqrt(x*x+y*y);
		max = (max > displacement[j]) ? max:displacement[j];
		average += displacement[j]; 
		printf("%.2f ",displacement[j]);
	}
	printf("\nMaximum displacement is %.2f", max);
	printf("\nAverage displacement is %.2f",average/1000);
	getchar();
	return 0;
}
